packages/terra-infinite-list/examples/index-examples/InfiniteListExample.jsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import IconInformation from 'terra-icon/lib/icon/IconInformation';
+import LoadingOverlay from 'terra-overlay/lib/LoadingOverlay';
+import OverlayContainer from 'terra-overlay/lib/OverlayContainer';
+import Arrange from 'terra-arrange';
+import classNames from 'classnames/bind';
+import styles from './InfiniteListExample.scss';
+import InfiniteList from '../../lib/InfiniteList';
+
+const cx = classNames.bind(styles);
+
+class InfiniteListExample extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.addMoreData = this.debounce(this.addMoreData.bind(this), 1000);
+    this.handleOnChange = this.handleOnChange.bind(this);
+
+    this.state = { stillLoading: true, numberOfPages: 0, selectedIndexes: [] };
+  }
+
+  handleOnChange(event, index) {
+    this.setState({ selectedIndexes: [index] });
+  }
+
+  addMoreData() {
+    const newNumberOfPages = this.state.numberOfPages + 1;
+    if (newNumberOfPages > 10) {
+      return;
+    } else if (newNumberOfPages > 9) {
```

Comment:
Instead of calling setState twice, can we have a set of props we pass into setState?